tests/components/cambridge_audio/test_config_flow.py

```diff
@@ -0,0 +1,148 @@
+    properties={
+        "serial": "0020c2d8",
+        "hcv": "3764",
+        "software": "v022-a-151+a",
+        "model": "CXNv2",
+        "udn": "02680b5c-1320-4d54-9f7c-3cfe915ad4c3",
+    },
+)
+
+
+async def test_full_flow(
+    hass: HomeAssistant,
+    mock_stream_magic_client: AsyncMock,
+    mock_setup_entry: AsyncMock,
+) -> None:
+    """Test full flow."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+    )
+    await hass.async_block_till_done()
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "user"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {CONF_HOST: "192.168.20.218"},
+    )
+    await hass.async_block_till_done()
+    assert result["type"] is FlowResultType.CREATE_ENTRY
+    assert result["title"] == "Cambridge Audio CXNv2"
+    assert result["data"] == {
+        CONF_HOST: "192.168.20.218",
+    }
+    assert result["result"].unique_id == "0020c2d8"
+
+
+async def test_flow_errors(
+    hass: HomeAssistant,
+    mock_stream_magic_client: AsyncMock,
+    mock_setup_entry: AsyncMock,
+) -> None:
+    """Test flow errors."""
+    mock_stream_magic_client.connect.side_effect = StreamMagicError
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+    )
+    await hass.async_block_till_done()
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "user"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {CONF_HOST: "192.168.20.218"},
+    )
+    await hass.async_block_till_done()
+
+    assert result["type"] is FlowResultType.FORM
+    assert result["errors"] == {"base": "cannot_connect"}
+
+    mock_stream_magic_client.connect.side_effect = None
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {CONF_HOST: "192.168.20.218"},
+    )
+    await hass.async_block_till_done()
+    assert result["type"] is FlowResultType.CREATE_ENTRY
+
+
+async def test_duplicate(
+    hass: HomeAssistant,
+    mock_stream_magic_client: AsyncMock,
+    mock_setup_entry: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test duplicate flow."""
+    mock_config_entry.add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+    )
+    await hass.async_block_till_done()
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "user"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {CONF_HOST: "192.168.20.218"},
+    )
+    await hass.async_block_till_done()
+
+    assert result["type"] is FlowResultType.ABORT
+    assert result["reason"] == "already_configured"
+
+
+async def test_zeroconf_flow(
```

Comment:
I am missing the zeroconf test where we try to discover an already added device, and the zeroconf test where the device doesn't respond